projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/SupportedPythonVersions.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.deploy;
+
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Value;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import lombok.extern.slf4j.Slf4j;
+
+/** Handles operations related to supported python versions for data job deployments. */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SupportedPythonVersions {
+
+  @Value("#{${datajobs.deployment.supportedPythonVersions}}")
+  private HashMap<String, HashMap<String, String>> supportedPythonVersions;
+
+  @Value("${datajobs.vdk.image}")
+  private String vdkImage;
+
+  @Value("${datajobs.deployment.dataJobBaseImage:python:3.9-slim}")
+  private String deploymentDataJobBaseImage;
+
+  private static final String VDK_IMAGE = "vdkImage";
+
+  private static final String BASE_IMAGE = "baseImage";
+
+  /**
+   * Check if the python_version passed by the user is supported by the Control Service.
+   *
+   * @param python_version python version passed by the user.
+   * @return true if the version is supported, and false otherwise.
+   */
+  public boolean isPythonVersionSupported(String python_version) {
+    if (!supportedPythonVersions.isEmpty()) {
+      return supportedPythonVersions.containsKey(python_version);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Returns a string of the python versions supported by the Control Service, in the format: [3.7,
+   * 3.8, ...]. If the supportedPythonVersions configuration is not set, the method returns the
+   * default python version set in the deploymentDataJobBaseImage configuration property.
+   *
+   * @return A string of all python versions supported by the Control Service
+   */
+  public String getSupportedPythonVersions() {
```

Comment:
This looks strange. Why return a string and not a list ?

What is the purpose of this method . How is it goingot be used. I don't see it used anywhere

Reply:
Agreed with @tozka , this method should return a list of Strings.

Reply:
Also, I think we should define the use cases and the behavior. 

We always have a default python version so the supportedPythonVersions should not be empty right? Somehow we should connect the default version to the list of supported versions.

How about that (or something similar)?
datajobs.deployment.supportedPythonVersions={3.7: {vdkImage: 'some-test-img', baseImage: 'python:3.7-slim'}}
datajobs.deployment.defaultPythonVersion=3.7

And we definitely need validation during the CS startup.

Reply:
Updated the method to return a list of strings, instead of one big string.

Reply:
Did a complete refactor.